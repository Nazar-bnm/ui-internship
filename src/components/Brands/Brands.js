import React, { Component } from 'react';
import cx from 'classnames';

import HttpService from '../../service/HttpService/httpService';

import './Brands.scss';

export const CN = 'brand';

class BrandsAndShippingInfo extends Component {
  constructor(props) {
    super(props);

    this.state = {
      brandsList: []
    };
  }

  componentDidMount() {
    this.getBrandsImages();
  }

  async getBrandsImages() {
    const userAPI = new HttpService();

    try {
      const response = await userAPI.get(`${process.env.BASE_URL}/brands`);
      if (response && response.data) {
        this.setState({ brandsList: response.data });
      }
    } catch (error) {
      throw (new Error());
    }
  }

  renderBrands(brands) {
    return brands.map(({ id, image }) => (
      <li key={id} className={`col-2 ${CN}__list-item`}>
        <img className={`${CN}__list-photo`} alt="brand logo" src={image} />
      </li>
    ));
  }

  render() {
    const { brandsList } = this.state;

    return (
      <div className="content">
        <h3 className={`${CN}__title`}>Top Brands</h3>
        <div className={`container ${cx(CN)}`}>
          <ul className={`${CN}__list`}>
            {this.renderBrands(brandsList)}
          </ul>
        </div>
      </div>
    );
  }
}

export default BrandsAndShippingInfo;
