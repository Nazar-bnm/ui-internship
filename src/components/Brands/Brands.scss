@import '../../scss/global_styles';

.brand {
  &__title {
    @include title-font;
  }

  &__list {
    display: flex;
    list-style: none;
    width: 100%;
    align-items: center;
    justify-content: space-between;
    margin-top: 30px;
    margin-bottom: 30px;

    &-item {
      cursor: pointer;
      padding: 0 17px;
    }

    &-photo {
      width: 100%;
      transition: all .3s ease-in-out;
      filter: opacity(.3);
    
      &:hover {
        transform: scale(1.2);
      }
    }
  }

  &__container {
    @extend .container;
    justify-content: center;
  }
}

@media screen and (max-width: 768px) {
  .brand__list-item {
    padding: 0 10px;
  }
}

@media screen and (max-width: 480px) {
  .brand {
    .col-2 {
      width: 35%;
    }

    &__list {
      justify-content: space-around;

      &-item {
        &:nth-of-type(1n+3) {
          display: none;
        }
      }
    }
  }
}
