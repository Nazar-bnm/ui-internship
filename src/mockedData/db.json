{
  "employees": [
    {
      "id": 1,
      "first_name": "Sebastian",
      "last_name": "Eschweiler",
      "email": "sebastian@codingthesmartway.com"
    },
    {
      "id": 2,
      "first_name": "Steve",
      "last_name": "Palmer",
      "email": "steve@codingthesmartway.com"
    },
    {
      "id": 3,
      "first_name": "Ann",
      "last_name": "Smith",
      "email": "ann@codingthesmartway.com"
    }
  ],
  "what-is-new" : [
    {
      "id": 1,
      "photo": "https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg",
      "title": "Black and White",
      "collection": "spring collection 2016",
      "category": "women"
    },
    {
      "id": 2,
      "photo": "https://www.next.ua/nxtcms/resource/image/1233600/portrait_ratio1x1/525/525/69ca86774e073288bc792479650a14cc/Jn/all-baby-girl.jpg",
      "title": "Color Summer",
      "collection": "spring collection 2016",
      "category": "kids"
    },
    {
      "id": 3,
      "photo": "https://www.next.ua/nxtcms/resource/image/1915752/portrait_ratio1x1/525/525/1cf45576cf1de5e2bab187220d4ebeb0/ds/g83-mb-sh819338-lm-357.jpg",
      "title": "Vintage For Him",
      "collection": "spring collection 2016",
      "category": "men"
    }
  ],
  "most-popular": [
    {
      "id": 1,
      "image": "https://i.dailymail.co.uk/1s/2018/12/15/12/7458890-6498875-image-a-2_1544876240893.jpg",
      "label": "new",
      "title": "Black dress",
      "price": "99,50$",
      "sizes": "S - M - L"
    },
    {
      "id": 2,
      "image": "https://i.pinimg.com/564x/99/5d/f3/995df3032c828381332af1dea4e5c590.jpg",
      "label": "sale",
      "title": "Red evening dress",
      "price": "179.00$",
      "sizes": "S - M - L"
    },
    {
      "id": 3,
      "image": "https://i.redd.it/grla2qtpa5721.jpg",
      "label": "",
      "title": "Fit & flare dress",
      "price": "99.00$",
      "sizes": "S - M - L"
    }
  ]
}
