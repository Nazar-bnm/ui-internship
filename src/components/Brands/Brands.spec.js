import React from 'react';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import { mount } from 'enzyme';

import Brands from './Brands';
import mockedData from '@/mockedDataForTests';

describe('<Brands />', () => {
  let wrapper;
  const { mockedBrands } = mockedData;

  beforeEach(() => {
    wrapper = mount(<Brands />);
    const mock = new MockAdapter(axios);

    mock.onGet(`${process.env.BASE_URL}/brands`).reply(200, mockedBrands);
  });

  test('should check `componentDidMount()`', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getBrandsImages');
    instance.componentDidMount();
    expect(instance.getBrandsImages).toHaveBeenCalledTimes(1);
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
