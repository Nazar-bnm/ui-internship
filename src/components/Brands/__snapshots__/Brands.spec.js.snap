// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Brands /> should match snapshot 1`] = `
<BrandsAndShippingInfo>
  <div
    className="content"
  >
    <h3
      className="brand__title"
    >
      Top Brands
    </h3>
    <div
      className="container brand"
    >
      <ul
        className="brand__list"
      />
    </div>
  </div>
</BrandsAndShippingInfo>
`;
